pkg/engine/engine.go

```diff
@@ -28,11 +28,11 @@ import (
 var ErrNotSupported = errors.New("feature not supported in new query engine")
 
 // New creates a new instance of the query engine that implements the [logql.Engine] interface.
-func New(opts logql.EngineOpts, bucket objstore.Bucket, limits logql.Limits, reg prometheus.Registerer, logger log.Logger) *QueryEngine {
+func New(opts logql.EngineOpts, cfg metastore.StorageConfig, bucket objstore.Bucket, limits logql.Limits, reg prometheus.Registerer, logger log.Logger) *QueryEngine {
 	var ms metastore.Metastore
 	if bucket != nil {
 		metastoreBucket := objstore.NewPrefixedBucket(bucket, opts.CataloguePath)
```

Comment:
You might want to get rid of this prefixed bucket now. I tusees the CataloguePath which is set to "index/v0" when reading from indexes are enabled. Since you use a per-query prefix passed to the metastore, this will build the prefix twice.